### PR TITLE
Vb 3308 additional tests

### PR DESCRIPTION
**Pull request by anunagula3**

### CODE REVIEW

File: src/test/resources/features/Sessions.feature [hunk 1]
```diff
@@ -299,17 +245,23 @@ Feature: Session template based tests
...
     And Im on "Manage prison visits - Search for a prisoner" page
     When I enter "<prisonNumber>" to search for a prison
     And click on search button
     Then I choose prison from search results
-    Then Im on "Manage prison visits - Vsip_prisoner14, Do Not Use" page
-    And I check Female category for that prisoner
+    Then Im on "Manage prison visits - Vsip_prisoner13, Do Not Use" page
+    And I click on Book a visit button
+    Then Im on "Manage prison visits - Select visitors from the prisoner’s approved visitor list" page
+    And I select a visitoor form the list from Bristol establishment
```
Comment (dhirajshettymoj) [inline src/test/resources/features/Sessions.feature:258]: minor typo on visitor

File: src/test/resources/features/Sessions.feature [hunk 2]
```diff
@@ -256,6 +188,13 @@ Feature: Session template based tests
     Then I choose prison from search results
     Then Im on "Manage prison visits - Vsip_prisoner12, Do Not Use" page
     And I check enhanced incentive level for that prisoner
+    And I click on Book a visit button
+    Then Im on "Manage prison visits - Select visitors from the prisoner’s approved visitor list" page
+    And I select a visitr form the list from Bristol establishment
```
Comment (dhirajshettymoj) [inline src/test/resources/features/Sessions.feature:193]: minor typo on visitor

File: src/test/resources/features/Sessions.feature [hunk 3]
```diff
@@ -24,41 +24,13 @@ Feature: Session template based tests
...
-    Then Im on "Manage prison visits - Manage prison visits" page
-    And I click on Book a visit option
-    And Im on "Manage prison visits - Search for a prisoner" page
-    When I enter "<prisonNumber>" to search for a prison
-    And click on search button
-    Then I choose prison from search results
-    Then Im on "Manage prison visits - Vsip_prisoner12, Do Not Use" page
-    And I select last booked visit reference
-    And I check "open" session is booked
+    And I check only Open time slots is available
```
[the remark this answers is not in the document]
Comment (anunagula3) [reply src/test/resources/features/Sessions.feature:27]: These are Individual Tests; all Tests scripted is Robust & totally independent. Can be executed on Independent basis.

File: src/test/java/uk/gov/justice/digital/hmpps/vsip/pages/SelectTimeSlotPage.java [hunk 4]
```diff
@@ -67,4 +67,68 @@ public void clickNextEnabledElement() {
             }
         }
     }
+
+    public void checkOnlyOpenTimeSlots() {
+        methodsService.isElementDisplayed("xpath","//p[contains(text(), '150 tables available')]");
```
[the remark this answers is not in the document]
Comment (anunagula3) [reply src/test/java/uk/gov/justice/digital/hmpps/vsip/pages/SelectTimeSlotPage.java:72]: Yes as discussed capturing Day will be written on run-time. I'll do more investigation over seeing the Test Execution for next few days & re-visit this

File: src/test/resources/features/Sessions.feature [hunk 5]
```diff
@@ -87,128 +59,81 @@ Feature: Session template based tests
...
-
     Given I navigate to VSIP service
     And Im on "HMPPS Digital Services - Sign in" page
     And I enter "<userName>"
     And I enter the "<password>"
     And click on submit button
     Then Im on "Manage prison visits - Manage prison visits" page
     And I select change establishment
-    And I change the establishmnet to Bristol
+    And I change the establishmnet to Drake Hall
```
[the remark this answers is not in the document]
Comment (anunagula3) [reply src/test/resources/features/Sessions.feature:80]: This is Done intentionally to differentiate the Test Steps; this is less complex Automation Test approach.

File: src/test/resources/features/Sessions.feature [hunk 6]
```diff
@@ -87,128 +59,81 @@ Feature: Session template based tests
...
```
[the remark this answers is not in the document]
Comment (anunagula3) [reply src/test/resources/features/Sessions.feature:116]: same as above

File: src/test/resources/features/Sessions.feature (hunk 1 above)
Comment (anunagula3) [reply src/test/resources/features/Sessions.feature:258]: same as above

File: src/test/resources/features/Sessions.feature (hunk 2 above)
Comment (anunagula3) [reply src/test/resources/features/Sessions.feature:193]: same as above

File: src/test/resources/features/Sessions.feature (hunk 5 above)
[the remark this answers is not in the document]
Comment (anunagula3) [reply src/test/resources/features/Sessions.feature:80]: Fixed

**Review by psoleckimoj — APPROVED** [review]
LGTM but @dhirajshettymoj will still need to check it covers what he wanted.

**Pull request merged**